lib/common/common.go

```diff
@@ -30,3 +45,9 @@ type ErrSeveralImages struct {
 func (e *ErrSeveralImages) Error() string {
 	return e.Msg
 }
+
+// ParseDockerURL takes a Docker URL and returns a ParsedDockerURL with its
+// index URL, image name, and tag.
+func ParseDockerURL(arg string) (*types.ParsedDockerURL, error) {
```

Comment:
This is troublesome - the returned result is a part of internal package, so the user of the `ParseDockerURL` function can use it directly, but cannot write a function that takes the `*types.ParsedDockerURL`, because the user will not be able to import the internal package…

I think that putting the `ParsedDockerURL` struct in the `common` package will end up in cyclic dependencies again, so consider adding `type ParsedDockerURL types.ParsedDockerURL` somewhere in the `common` package and do some casting in the `ParseDockerURL` function.